minijinja/src/value/mod.rs

```diff
@@ -879,9 +935,23 @@ impl Value {
                 items.len(),
             ),
             ValueRepr::Dynamic(ref obj) => {
-                let attrs = obj.attributes().map(Value::from).collect::<Vec<_>>();
-                let attr_count = attrs.len();
-                (ValueIteratorState::Seq(0, Arc::new(attrs)), attr_count)
+                match obj.behavior() {
+                    ObjectBehavior::Basic => (ValueIteratorState::Empty, 0),
+                    ObjectBehavior::Seq(s) => {
+                        // TODO: lazy iteration?
+                        let mut items = vec![];
+                        for idx in 0..s.len() {
+                            items.push(s.get(idx).unwrap_or(Value::UNDEFINED));
+                        }
+                        (ValueIteratorState::Seq(0, Arc::new(items)), s.len())
+                    }
+                    ObjectBehavior::Struct(s) => {
+                        // TODO: lazy iteration?
+                        let attrs = s.fields().map(Value::from).collect::<Vec<_>>();
+                        let attr_count = s.len();
+                        (ValueIteratorState::Seq(0, Arc::new(attrs)), attr_count)
+                    }
+                }
```

Comment:
I don't know where this is used, but as the comments indicate, this is really much more expensive than I would have expected

Reply:
Yeah, it's definitely worth addressing but I don't think it blocks at least the initial implementation. The fact that you can't hold on to a `&dyn AsSeq` is the issue here. One way without unsafe would be to have the iterator state reborrow the `AsSeq` on every iteration but I found that to be surprising.